components/api_version/src/lib.rs

```diff
@@ -593,4 +618,163 @@ mod tests {
             );
         })
     }
+
+    #[test]
+    fn test_raw_key_convert() {
+        let timestamp = 30;
+        let apiv1_keys = vec![
+            b""[..].to_owned(),
+            b"abc"[..].to_owned(),
+            b"api_ver_test"[..].to_owned(),
+        ];
+        let apiv2_keys: Vec<Vec<u8>> = apiv1_keys
+            .clone()
+            .into_iter()
+            .map(|key| {
+                let mut v2_key = key;
+                v2_key.insert(0, RAW_KEY_PREFIX);
+                ApiV2::encode_raw_key_owned(v2_key, Some(TimeStamp::from(timestamp))).into_encoded()
+            })
+            .collect();
+        // src_api_ver, dst_api_ver, src_data, dst_data
+        let test_cases = vec![
+            (ApiVersion::V1, ApiVersion::V2, &apiv1_keys, &apiv2_keys),
+            (ApiVersion::V1ttl, ApiVersion::V2, &apiv1_keys, &apiv2_keys),
+            (ApiVersion::V2, ApiVersion::V1, &apiv2_keys, &apiv1_keys),
+            (ApiVersion::V2, ApiVersion::V1ttl, &apiv2_keys, &apiv1_keys),
+        ];
+        for i in 0..apiv1_keys.len() {
+            for case in &test_cases {
```

Comment:
We could use a pattern matching here, match a pair `(first, second)`

Reply:
Ok, I will change it, it's more clear.